lesson7/main.js

```diff
@@ -1,34 +1,54 @@
 "use strict";
 
+const wrapper = document.getElementById("js-wrapper");
 const ul = document.getElementById("js-lists");
 const fragment = document.createDocumentFragment();
-const fetchedData = new Promise((resolve, reject) => {
-  const listItems = [
-    { to: "bookmark.html", img: "1.png", alt: "画像1", text: "ブックマーク" },
-    { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" }
-  ];
-  setTimeout(() => {
-    resolve(listItems);
-  }, 3000);
-});
-
-function createNewList(data) {
-  data.forEach((key) => {
-    const li = document.createElement("li");
-    const a = document.createElement("a");
-    const img = document.createElement("img");
-
-    a.href = `/${key.to}`;
-    a.textContent = key.text;
-    img.src = key.img;
-    img.alt = key.alt;
-
-    fragment
-      .appendChild(li)
-      .appendChild(a)
-      .insertAdjacentElement("afterbegin", img);
+const loader = document.createElement("div");
+
+const loading = () => {
+  const loaderImage = document.createElement("img");
+  loader.classList.add("loading");
+  loaderImage.src = "img/loading-circle.gif";
+  wrapper.appendChild(loader).appendChild(loaderImage);
+};
+
+const loaded = () => {
+  loader.classList.add("loaded");
+};
```

Comment:
This one is only used in functions, so I think it is better to define it where it will be used, instead of defining it globally.
`const loader = document.createElement("div");`


IMO
```suggestion
const loading = () => {
  const loader = document.createElement("div");
  const loaderImage = document.createElement("img");
  loader.id="loader"
  loader.classList.add("loading");
  loaderImage.src = "img/loading-circle.gif";
  wrapper.appendChild(loader).appendChild(loaderImage);
};
const loaded = () => {
  const loader = document.getElementById("loader");
  loader.classList.add("loaded");
};
```

Reply:
Oh I see! I moved `const loader = document.createElement("div");` in global because `loaded function` also uses this `loader`, but actually I felt it's wired...
But now I understand how to control it! I appreciated.